dora/core/client/fs/src/main/java/alluxio/client/file/PositionReadFileInStream.java

```diff
@@ -192,75 +198,87 @@ public int read(byte[] b, int off, int len) throws IOException {
 
   @Override
   public int read(ByteBuffer byteBuffer, int off, int len) throws IOException {
+    LOG.error("Reading len={} mPositionReader={}", len, mPositionReader);
     byteBuffer.position(off).limit(off + len);
     mCache.addTrace(mPos, len);
     int totalBytesRead = 0;
     int bytesReadFromCache = mCache.fillWithCache(mPos, byteBuffer);
     totalBytesRead += bytesReadFromCache;
     mPos += bytesReadFromCache;
     if (!byteBuffer.hasRemaining()) {
+      Metrics.BYTES_READ_ALLUXIO.inc(totalBytesRead);
       return totalBytesRead;
     }
     int bytesPrefetched = mCache.prefetch(mPositionReader, mPos, byteBuffer.remaining());
     if (bytesPrefetched < 0) {
       if (totalBytesRead == 0) {
         return -1;
       }
+      Metrics.BYTES_READ_ALLUXIO.inc(totalBytesRead);
```

Comment:
what's the definition of Bytes read alluxio?
if the data is provided by the cache of this class, is it read from alluxio?

Reply:
Similar to write operation, we have two metrics:
read from /write to Alluxio, means data is read from worker or written to worker.
Read from /write to UFS, means data is (directly) read from UFS or written directly to UFS. (This is read operation in client fallback scenario, or write operation in which case the data is also written to UFS).